com.zsmartsystems.zigbee.console/src/main/java/com/zsmartsystems/zigbee/console/ZigBeeConsoleCommandsSupportedCommand.java

```diff
@@ -0,0 +1,111 @@
+import com.zsmartsystems.zigbee.zcl.ZclCluster;
+import com.zsmartsystems.zigbee.zcl.ZclCommand;
+
+/**
+ * Console command that prints the commands that are supported by a given cluster.
+ *
+ * @author Henning Sudbrock
+ */
+public class ZigBeeConsoleCommandsSupportedCommand extends ZigBeeConsoleAbstractCommand {
+
+    @Override
+    public String getCommand() {
+        return "cmdsupported";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Check what commands are supported within a cluster, optionally restricted to generated / received commands";
+    }
+
+    @Override
+    public String getSyntax() {
+        return "ENDPOINT CLUSTER [gen|rcv]";
+    }
+
+    @Override
+    public String getHelp() {
+        return "";
+    }
+
+    @Override
+    public void process(ZigBeeNetworkManager networkManager, String[] args, PrintStream out)
+            throws IllegalArgumentException, InterruptedException, ExecutionException {
+        if (args.length < 3 || args.length > 4) {
+            throw new IllegalArgumentException("Invalid number of arguments");
+        }
+
+        final ZigBeeEndpoint endpoint = getEndpoint(networkManager, args[1]);
+        final Integer clusterId = parseCluster(args[2]);
+        ZclCluster cluster = getCluster(endpoint, clusterId);
+
+        if (cluster == null) {
+            out.println("Cluster not found");
+            return;
+        }
+
+        if (showGenerated(args)) {
+            if (cluster.discoverCommandsGenerated(false).get()) {
+                out.println("Supported generated commands for " + cluster.getClusterName() + " cluster "
+                        + printClusterId(cluster.getClusterId()));
+                printCommands(out, cluster, cluster.getSupportedCommandsGenerated());
+
+            } else {
+                out.println("Failed to retrieve supported generated commands");
+            }
+            out.println();
+        }
+
+        if (showReceived(args)) {
+            if (cluster.discoverCommandsReceived(false).get()) {
+                out.println("Supported received commands for " + cluster.getClusterName() + " cluster "
+                        + printClusterId(cluster.getClusterId()));
+                printCommands(out, cluster, cluster.getSupportedCommandsReceived());
+            } else {
+                out.println("Failed to retrieve supported received commands");
+            }
+        }
+    }
+
+    private ZclCluster getCluster(ZigBeeEndpoint endpoint, Integer clusterId) {
+        ZclCluster result = endpoint.getInputCluster(clusterId);
+        if (result == null) {
+            result = endpoint.getOutputCluster(clusterId);
+        }
+        return result;
+    }
+
+    private void printCommands(PrintStream out, ZclCluster cluster, Set<Integer> commandIds) {
+        out.println("CommandId  Command");
+        for (Integer commandId : commandIds) {
+            out.print(" ");
```

Comment:
Why are there 3 print statement (2 prints, followed by the println)?

Reply:
Copied from `attsupported` command :) But you're right, this can be improved here, thanks.